src/middlewared/middlewared/plugins/account.py

```diff
@@ -842,7 +842,14 @@ def do_update(self, app, audit_callback, pk, data):
     def recreate_homedir_if_not_exists(self, user, group, mode):
         # sigh, nothing is stopping someone from removing the homedir
         # from the CLI so recreate the original directory in this case
-        if not os.path.isdir(user['home']):
+        if os.path.isdir(user['home']):
```

Comment:
I think we can simplify this a bit more. We're calling `os.path.isdir` and potentially `os.path.exists` so we could remove the double stat call by doing something like this:
```
try:
    st = self.middleware.call_sync('filesystem.stat', user['home'])
except CallError as e:
    if e.errno == errno.ENOENT:
        pass
    else:
        # maybe log something here? or do we raise?
else:
    if st['type'] == 'DIRECTORY':
        if st['realpath'].startswith('/mnt'):
            ...
    else:
        raise CallError(f'{user["home"]!r} already exists and is not a directory')

Reply:
Wonder what @anodos325 thinks about this. I could be wrong :smile:

Reply:
Yeah, that's probably better. If we have stat output then we can also use it to determine whether we _actually_ need to chown the directory.